eva-accession-remapping/src/test/java/uk/ac/ebi/eva/accession/remapping/batch/io/VariantContextWriterTest.java

```diff
@@ -74,59 +136,80 @@ public void writeRsId() throws Exception {
+        String specialProject = "project2";
+        File output = assertWriteVcf(outputFolder, buildVariant("1", 1000, "C", "A", null, specialProject));
 
         long variantCount = forEachVcfDataLine(output, (String[] columns) -> {
             assertEquals(COLUMNS_IN_VCF_WITHOUT_SAMPLES, columns.length);
-            assertEquals(".", columns[VCF_INFO_COLUMN]);
+            assertInfo(null, specialProject, columns[VCF_INFO_COLUMN]);
         });
         assertEquals(1, variantCount);
     }
+    @Test
+    public void writeProjectAndRs() throws Exception {
+        File outputFolder = temporaryFolder.newFolder();
+        String specialProject = "project2";
+        long rsId = 123L;
+        File output = assertWriteVcf(outputFolder, buildVariant("1", 1000, "C", "A", rsId, specialProject));
 
-    private File assertWriteVcf(File outputFolder, SubmittedVariantEntity... variants) throws Exception {
-        Path reportPath = ReportPathResolver.getEvaReportPath(outputFolder.getAbsolutePath(), REFERENCE_ASSEMBLY);
-        VariantContextWriter writer = new VariantContextWriter(reportPath, REFERENCE_ASSEMBLY);
-        writer.open(null);
-
-        SubmittedVariantToVariantContextProcessor variantToVariantContextProcessor =
-                new SubmittedVariantToVariantContextProcessor();
-
-        List<VariantContext> variantContexts =
-                Stream.of(variants).map(variantToVariantContextProcessor::process).collect(Collectors.toList());
-        writer.write(variantContexts);
-
-        writer.close();
+        long variantCount = forEachVcfDataLine(output, (String[] columns) -> {
+            assertEquals(COLUMNS_IN_VCF_WITHOUT_SAMPLES, columns.length);
+            assertInfo(rsId, specialProject, columns[VCF_INFO_COLUMN]);
+        });
+        assertEquals(1, variantCount);
+    }
 
-        File output = writer.getOutput();
-        assertTrue(output.exists());
 
-        return output;
-    }
+    @Test
+    public void writeSeveralVariants() throws Exception {
+        File outputFolder = temporaryFolder.newFolder();
+        String specialProject = "project2";
+        long rsId = 123L;
+        File output = assertWriteVcf(outputFolder,
+                                     buildVariant("1", 1000, "C", "A", rsId, specialProject),
+                                     buildVariant("2", 1000, "C", "A", rsId, specialProject),
+                                     buildVariant("3", 1000, "C", "A", rsId, specialProject),
+                                     buildVariant("4", 1000, "C", "A", rsId, specialProject));
 
-    private SubmittedVariantEntity buildVariant(String chr, long start, String ref, String alt, Long rs) {
-        return new SubmittedVariantEntity(1L, "hash1", REFERENCE_ASSEMBLY, 9606, "project1", chr, start, ref, alt, rs,
-                                          false, false, false, false, 1);
+        final Long[] expectedChr = {1L};
```

Comment:
Why should this be an array?

Reply:
apparently lambda closures can only take `final` variables, and the contents of a `final` array can change. Intellij suggested this to get rid of the compilation error.